Fix stackInstert() in algo.c

Mathazzar opened the pull request:
Modified stackInstert() to return a bool that indicates if the node was added to the stack successfully for debugging and additional path-choosing options.
Fixed final step of for-loop in stackInstert() to copy tempArr to nodeCurrent instead.

Mathazzar committed:
Fix stackInstert() in algo.c

Modified stackInstert() to return a bool that indicates if the node was added to the stack successfully for debugging and additional path-choosing options.
Fixed final step of for-loop in stackInster() to copy tempArr to nodeCurrent instead.

squidKnight reviewed (commented):
I just tested it with the looping version in Mathazzar-patch-1. For some reason it does not print part of the stack after a certain number of looping in the maze (you can really see the stack decrease in size as it prints for maze 5) I attached a screenshot to show you what I mean (hopefully, it shows up okay here):
![image](https://user-images.githubusercontent.com/33043582/78615837-88685380-7840-11ea-90f7-73e1d6c13b05.png)

I'm not sure why this is happening exactly, because your logic looks fine. Maybe it's an issue with the number of loops in stackInsert()? I had a lot of trouble creating stackInsert(), so it might very well already have fatal logic error(s) that your changes exposed. Where do you think the issue might be?

Mathazzar committed:
Add notes to algo.c

Add reminders to prevent logic errors in stackInsert() when 2 nodes are directly next to eachother.
